File: services/src/main/java/com/liferay/mobile/sdk/v7/socialactivity/SocialActivityService.java
package com.liferay.mobile.sdk.v7.socialactivity;

import com.liferay.mobile.sdk.Call;
import com.liferay.mobile.sdk.annotation.Param;
import com.liferay.mobile.sdk.annotation.Path;
import java.lang.Integer;
import java.lang.String;
import org.json.JSONArray;
import org.json.JSONObject;

@Path("/socialactivity")
public interface SocialActivityService {
  @Path("/get-activities")
  Call<JSONArray> getActivities(@Param(name = "classNameId") long classNameId, @Param(name = "start") int start, @Param(name = "end") int end);

  @Path("/get-activities")
  Call<JSONArray> getActivities(@Param(name = "className") String className, @Param(name = "start") int start, @Param(name = "end") int end);

  @Path("/get-activities")
  Call<JSONArray> getActivities(@Param(name = "mirrorActivityId") long mirrorActivityId, @Param(name = "classNameId") long classNameId, @Param(name = "classPK") long classPK, @Param(name = "start") int start, @Param(name = "end") int end);

  @Path("/get-activities")
  Call<JSONArray> getActivities(@Param(name = "mirrorActivityId") long mirrorActivityId, @Param(name = "className") String className, @Param(name = "classPK") long classPK, @Param(name = "start") int start, @Param(name = "end") int end);

  @Path("/get-activities-count")
  Call<Integer> getActivitiesCount(@Param(name = "className") String className);

  @Path("/get-activities-count")
  Call<Integer> getActivitiesCount(@Param(name = "classNameId") long classNameId);

  @Path("/get-activities-count")
  Call<Integer> getActivitiesCount(@Param(name = "mirrorActivityId") long mirrorActivityId, @Param(name = "className") String className, @Param(name = "classPK") long classPK);

  @Path("/get-activities-count")
  Call<Integer> getActivitiesCount(@Param(name = "mirrorActivityId") long mirrorActivityId, @Param(name = "classNameId") long classNameId, @Param(name = "classPK") long classPK);

  @Path("/get-activity")
  Call<JSONObject> getActivity(@Param(name = "activityId") long activityId);

  @Path("/get-activity-set-activities")
  Call<JSONArray> getActivitySetActivities(@Param(name = "activitySetId") long activitySetId, @Param(name = "start") int start, @Param(name = "end") int end);

  @Path("/get-group-activities")
  Call<JSONArray> getGroupActivities(@Param(name = "groupId") long groupId, @Param(name = "start") int start, @Param(name = "end") int end);

  @Path("/get-group-activities-count")
  Call<Integer> getGroupActivitiesCount(@Param(name = "groupId") long groupId);

  @Path("/get-group-users-activities")
  Call<JSONArray> getGroupUsersActivities(@Param(name = "groupId") long groupId, @Param(name = "start") int start, @Param(name = "end") int end);

  @Path("/get-group-users-activities-count")
  Call<Integer> getGroupUsersActivitiesCount(@Param(name = "groupId") long groupId);

  @Path("/get-mirror-activity")
  Call<JSONObject> getMirrorActivity(@Param(name = "mirrorActivityId") long mirrorActivityId);

  @Path("/get-organization-activities")
  Call<JSONArray> getOrganizationActivities(@Param(name = "organizationId") long organizationId, @Param(name = "start") int start, @Param(name = "end") int end);

  @Path("/get-organization-activities-count")
  Call<Integer> getOrganizationActivitiesCount(@Param(name = "organizationId") long organizationId);

  @Path("/get-organization-users-activities")
  Call<JSONArray> getOrganizationUsersActivities(@Param(name = "organizationId") long organizationId, @Param(name = "start") int start, @Param(name = "end") int end);

  @Path("/get-organization-users-activities-count")
  Call<Integer> getOrganizationUsersActivitiesCount(@Param(name = "organizationId") long organizationId);

  @Path("/get-relation-activities")
  Call<JSONArray> getRelationActivities(@Param(name = "userId") long userId, @Param(name = "start") int start, @Param(name = "end") int end);

  @Path("/get-relation-activities")
  Call<JSONArray> getRelationActivities(@Param(name = "userId") long userId, @Param(name = "type") int type, @Param(name = "start") int start, @Param(name = "end") int end);

  @Path("/get-relation-activities-count")
  Call<Integer> getRelationActivitiesCount(@Param(name = "userId") long userId);

  @Path("/get-relation-activities-count")
  Call<Integer> getRelationActivitiesCount(@Param(name = "userId") long userId, @Param(name = "type") int type);

  @Path("/get-user-activities")
  Call<JSONArray> getUserActivities(@Param(name = "userId") long userId, @Param(name = "start") int start, @Param(name = "end") int end);

  @Path("/get-user-activities-count")
  Call<Integer> getUserActivitiesCount(@Param(name = "userId") long userId);

  @Path("/get-user-groups-activities")
  Call<JSONArray> getUserGroupsActivities(@Param(name = "userId") long userId, @Param(name = "start") int start, @Param(name = "end") int end);

  @Path("/get-user-groups-activities-count")
  Call<Integer> getUserGroupsActivitiesCount(@Param(name = "userId") long userId);

  @Path("/get-user-groups-and-organizations-activities")
  Call<JSONArray> getUserGroupsAndOrganizationsActivities(@Param(name = "userId") long userId, @Param(name = "start") int start, @Param(name = "end") int end);

  @Path("/get-user-groups-and-organizations-activities-count")
  Call<Integer> getUserGroupsAndOrganizationsActivitiesCount(@Param(name = "userId") long userId);

  @Path("/get-user-organizations-activities")
  Call<JSONArray> getUserOrganizationsActivities(@Param(name = "userId") long userId, @Param(name = "start") int start, @Param(name = "end") int end);

  @Path("/get-user-organizations-activities-count")
  Call<Integer> getUserOrganizationsActivitiesCount(@Param(name = "userId") long userId);
}

File: windows/Liferay.SDK/Service/V62/SocialActivity/SocialActivityService.cs
//------------------------------------------------------------------------------
// <copyright file="SocialActivityService.cs">
//    Copyright (c) 2014-present Andrea Di Giorgi
//
//    Permission is hereby granted, free of charge, to any person obtaining a
//    copy of this software and associated documentation files (the "Software"),
//    to deal in the Software without restriction, including without limitation
//    the rights to use, copy, modify, merge, publish, distribute, sublicense,
//    and/or sell copies of the Software, and to permit persons to whom the
//    Software is furnished to do so, subject to the following conditions:
//
//    The above copyright notice and this permission notice shall be included in
//    all copies or substantial portions of the Software.
//
//    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
//    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
//    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
//    THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
//    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
//    FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
//    DEALINGS IN THE SOFTWARE.
// </copyright>
// <author>Andrea Di Giorgi</author>
// <website>https://github.com/Ithildir/liferay-sdk-builder-windows</website>
//------------------------------------------------------------------------------

using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Threading.Tasks;

namespace Liferay.SDK.Service.V62.SocialActivity
{
	public class SocialActivityService : ServiceBase
	{
		public SocialActivityService(ISession session)
			: base(session)
		{
		}

		public async Task<IEnumerable<dynamic>> GetActivitiesAsync(string className, int start, int end)
		{
			var _parameters = new JsonObject();

			_parameters.Add("className", className);
			_parameters.Add("start", start);
			_parameters.Add("end", end);

			var _command = new JsonObject()
			{
				{ "/socialactivity/get-activities", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IEnumerable<dynamic>)_obj;
		}

		public async Task<IEnumerable<dynamic>> GetActivitiesAsync(long classNameId, int start, int end)
		{
			var _parameters = new JsonObject();

			_parameters.Add("classNameId", classNameId);
			_parameters.Add("start", start);
			_parameters.Add("end", end);

			var _command = new JsonObject()
			{
				{ "/socialactivity/get-activities", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IEnumerable<dynamic>)_obj;
		}

		public async Task<IEnumerable<dynamic>> GetActivitiesAsync(long mirrorActivityId, string className, long classPK, int start, int end)
		{
			var _parameters = new JsonObject();

			_parameters.Add("mirrorActivityId", mirrorActivityId);
			_parameters.Add("className", className);
			_parameters.Add("classPK", classPK);
			_parameters.Add("start", start);
			_parameters.Add("end", end);

			var _command = new JsonObject()
			{
				{ "/socialactivity/get-activities", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IEnumerable<dynamic>)_obj;
		}

		public async Task<IEnumerable<dynamic>> GetActivitiesAsync(long mirrorActivityId, long classNameId, long classPK, int start, int end)
		{
			var _parameters = new JsonObject();

			_parameters.Add("mirrorActivityId", mirrorActivityId);
			_parameters.Add("classNameId", classNameId);
			_parameters.Add("classPK", classPK);
			_parameters.Add("start", start);
			_parameters.Add("end", end);

			var _command = new JsonObject()
			{
				{ "/socialactivity/get-activities", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IEnumerable<dynamic>)_obj;
		}

		public async Task<long> GetActivitiesCountAsync(string className)
		{
			var _parameters = new JsonObject();

			_parameters.Add("className", className);

			var _command = new JsonObject()
			{
				{ "/socialactivity/get-activities-count", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (long)_obj;
		}

		public async Task<long> GetActivitiesCountAsync(long classNameId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("classNameId", classNameId);

			var _command = new JsonObject()
			{
				{ "/socialactivity/get-activities-count", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (long)_obj;
		}

		public async Task<long> GetActivitiesCountAsync(long mirrorActivityId, string className, long classPK)
		{
			var _parameters = new JsonObject();

			_parameters.Add("mirrorActivityId", mirrorActivityId);
			_parameters.Add("className", className);
			_parameters.Add("classPK", classPK);

			var _command = new JsonObject()
			{
				{ "/socialactivity/get-activities-count", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (long)_obj;
		}

		public async Task<long> GetActivitiesCountAsync(long mirrorActivityId, long classNameId, long classPK)
		{
			var _parameters = new JsonObject();

			_parameters.Add("mirrorActivityId", mirrorActivityId);
			_parameters.Add("classNameId", classNameId);
			_parameters.Add("classPK", classPK);

			var _command = new JsonObject()
			{
				{ "/socialactivity/get-activities-count", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (long)_obj;
		}

		public async Task<dynamic> GetActivityAsync(long activityId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("activityId", activityId);

			var _command = new JsonObject()
			{
				{ "/socialactivity/get-activity", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<IEnumerable<dynamic>> GetActivitySetActivitiesAsync(long activitySetId, int start, int end)
		{
			var _parameters = new JsonObject();

			_parameters.Add("activitySetId", activitySetId);
			_parameters.Add("start", start);
			_parameters.Add("end", end);

			var _command = new JsonObject()
			{
				{ "/socialactivity/get-activity-set-activities", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IEnumerable<dynamic>)_obj;
		}

		public async Task<IEnumerable<dynamic>> GetGroupActivitiesAsync(long groupId, int start, int end)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);
			_parameters.Add("start", start);
			_parameters.Add("end", end);

			var _command = new JsonObject()
			{
				{ "/socialactivity/get-group-activities", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IEnumerable<dynamic>)_obj;
		}

		public async Task<long> GetGroupActivitiesCountAsync(long groupId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);

			var _command = new JsonObject()
			{
				{ "/socialactivity/get-group-activities-count", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (long)_obj;
		}

		public async Task<IEnumerable<dynamic>> GetGroupUsersActivitiesAsync(long groupId, int start, int end)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);
			_parameters.Add("start", start);
			_parameters.Add("end", end);

			var _command = new JsonObject()
			{
				{ "/socialactivity/get-group-users-activities", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IEnumerable<dynamic>)_obj;
		}

		public async Task<long> GetGroupUsersActivitiesCountAsync(long groupId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);

			var _command = new JsonObject()
			{
				{ "/socialactivity/get-group-users-activities-count", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (long)_obj;
		}

		public async Task<dynamic> GetMirrorActivityAsync(long mirrorActivityId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("mirrorActivityId", mirrorActivityId);

			var _command = new JsonObject()
			{
				{ "/socialactivity/get-mirror-activity", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<IEnumerable<dynamic>> GetOrganizationActivitiesAsync(long organizationId, int start, int end)
		{
			var _parameters = new JsonObject();

			_parameters.Add("organizationId", organizationId);
			_parameters.Add("start", start);
			_parameters.Add("end", end);

			var _command = new JsonObject()
			{
				{ "/socialactivity/get-organization-activities", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IEnumerable<dynamic>)_obj;
		}

		public async Task<long> GetOrganizationActivitiesCountAsync(long organizationId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("organizationId", organizationId);

			var _command = new JsonObject()
			{
				{ "/socialactivity/get-organization-activities-count", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (long)_obj;
		}

		public async Task<IEnumerable<dynamic>> GetOrganizationUsersActivitiesAsync(long organizationId, int start, int end)
		{
			var _parameters = new JsonObject();

			_parameters.Add("organizationId", organizationId);
			_parameters.Add("start", start);
			_parameters.Add("end", end);

			var _command = new JsonObject()
			{
				{ "/socialactivity/get-organization-users-activities", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IEnumerable<dynamic>)_obj;
		}

		public async Task<long> GetOrganizationUsersActivitiesCountAsync(long organizationId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("organizationId", organizationId);

			var _command = new JsonObject()
			{
				{ "/socialactivity/get-organization-users-activities-count", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (long)_obj;
		}

		public async Task<IEnumerable<dynamic>> GetRelationActivitiesAsync(long userId, int start, int end)
		{
			var _parameters = new JsonObject();

			_parameters.Add("userId", userId);
			_parameters.Add("start", start);
			_parameters.Add("end", end);

			var _command = new JsonObject()
			{
				{ "/socialactivity/get-relation-activities", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IEnumerable<dynamic>)_obj;
		}

		public async Task<IEnumerable<dynamic>> GetRelationActivitiesAsync(long userId, int type, int start, int end)
		{
			var _parameters = new JsonObject();

			_parameters.Add("userId", userId);
			_parameters.Add("type", type);
			_parameters.Add("start", start);
			_parameters.Add("end", end);

			var _command = new JsonObject()
			{
				{ "/socialactivity/get-relation-activities", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IEnumerable<dynamic>)_obj;
		}

		public async Task<long> GetRelationActivitiesCountAsync(long userId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("userId", userId);

			var _command = new JsonObject()
			{
				{ "/socialactivity/get-relation-activities-count", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (long)_obj;
		}

		public async Task<long> GetRelationActivitiesCountAsync(long userId, int type)
		{
			var _parameters = new JsonObject();

			_parameters.Add("userId", userId);
			_parameters.Add("type", type);

			var _command = new JsonObject()
			{
				{ "/socialactivity/get-relation-activities-count", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (long)_obj;
		}

		public async Task<IEnumerable<dynamic>> GetUserActivitiesAsync(long userId, int start, int end)
		{
			var _parameters = new JsonObject();

			_parameters.Add("userId", userId);
			_parameters.Add("start", start);
			_parameters.Add("end", end);

			var _command = new JsonObject()
			{
				{ "/socialactivity/get-user-activities", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IEnumerable<dynamic>)_obj;
		}

		public async Task<long> GetUserActivitiesCountAsync(long userId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("userId", userId);

			var _command = new JsonObject()
			{
				{ "/socialactivity/get-user-activities-count", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (long)_obj;
		}

		public async Task<IEnumerable<dynamic>> GetUserGroupsActivitiesAsync(long userId, int start, int end)
		{
			var _parameters = new JsonObject();

			_parameters.Add("userId", userId);
			_parameters.Add("start", start);
			_parameters.Add("end", end);

			var _command = new JsonObject()
			{
				{ "/socialactivity/get-user-groups-activities", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IEnumerable<dynamic>)_obj;
		}

		public async Task<long> GetUserGroupsActivitiesCountAsync(long userId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("userId", userId);

			var _command = new JsonObject()
			{
				{ "/socialactivity/get-user-groups-activities-count", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (long)_obj;
		}

		public async Task<IEnumerable<dynamic>> GetUserGroupsAndOrganizationsActivitiesAsync(long userId, int start, int end)
		{
			var _parameters = new JsonObject();

			_parameters.Add("userId", userId);
			_parameters.Add("start", start);
			_parameters.Add("end", end);

			var _command = new JsonObject()
			{
				{ "/socialactivity/get-user-groups-and-organizations-activities", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IEnumerable<dynamic>)_obj;
		}

		public async Task<long> GetUserGroupsAndOrganizationsActivitiesCountAsync(long userId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("userId", userId);

			var _command = new JsonObject()
			{
				{ "/socialactivity/get-user-groups-and-organizations-activities-count", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (long)_obj;
		}

		public async Task<IEnumerable<dynamic>> GetUserOrganizationsActivitiesAsync(long userId, int start, int end)
		{
			var _parameters = new JsonObject();

			_parameters.Add("userId", userId);
			_parameters.Add("start", start);
			_parameters.Add("end", end);

			var _command = new JsonObject()
			{
				{ "/socialactivity/get-user-organizations-activities", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IEnumerable<dynamic>)_obj;
		}

		public async Task<long> GetUserOrganizationsActivitiesCountAsync(long userId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("userId", userId);

			var _command = new JsonObject()
			{
				{ "/socialactivity/get-user-organizations-activities-count", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (long)_obj;
		}
	}
}